src/microbe_stage/MicrobeHUD.cs

```diff
@@ -756,31 +757,86 @@ public void SendEditorButtonToTutorial(TutorialState tutorialState)
     /// </summary>
     private void UpdateNeededBars()
     {
-        var compounds = stage!.Player?.Compounds;
+        // if the player isn't in a colony than do everything the way its used to be done.
+        if (stage!.Player!.Colony == null)
+        {
+            var compounds = stage!.Player?.Compounds;
 
-        if (compounds?.HasAnyBeenSetUseful() != true)
-            return;
+            if (compounds?.HasAnyBeenSetUseful() != true)
+                return;
 
-        if (compounds.IsSpecificallySetUseful(oxytoxy))
-        {
-            agentsPanel.Show();
+            if (compounds.IsSpecificallySetUseful(oxytoxy))
+            {
+                agentsPanel.Show();
+            }
+            else
+            {
+                agentsPanel.Hide();
+            }
+
+            foreach (ProgressBar bar in compoundBars)
+            {
+                var compound = SimulationParameters.Instance.GetCompound(bar.Name);
+
+                if (compounds.IsUseful(compound))
+                {
+                    bar.Show();
+                }
+                else
+                {
+                    bar.Hide();
+                }
+            }
         }
         else
         {
-            agentsPanel.Hide();
-        }
+            // Stores the names of the Progressbars,
+            // then during the code if they are true they git to avoid flashing bars.
```

Comment:
Please reword this comment, I'm having a hard time trying to read and understand this.